storage/range.go

```diff
@@ -940,6 +939,8 @@ func (r *Range) EndTransaction(batch engine.Engine, args *proto.EndTransactionRe
 			reply.SetGoError(r.splitTrigger(batch, args.SplitTrigger))
 		} else if args.MergeTrigger != nil {
 			reply.SetGoError(r.mergeTrigger(batch, args.MergeTrigger))
+		} else if args.ChangeReplicasTrigger != nil {
+			// ChangeReplicasTrigger is processed on input, not output.
```

Comment:
Actually, I think here you'll want to actually call a changeReplicasTrigger method which could then set the updated replicas in the descriptor. You might want to modify ChangeReplicasTrigger protobuf to accept an UpdatedDesc for this purpose.